### ISSUE TITLE
`sortedUniq` does not return unique array if input is not sorted

### ISSUE THREAD

**Issue by potmo** [lodash/lodash#3305]
When using `sortedUniq` with a non sorted array the duplicates are not removed from the output. The documentation states that 

> This method is like `_.uniq` except that it's designed and optimized for sorted arrays.

that is wrong since it does not return a unique array if the input is not sorted.

```javascript
_.sortedUniq(["test1", "test2", "test3", "test4", "test1", "test2", "test3", "test4"])
``` 
returns `["test1", "test2", "test3", "test4", "test1", "test2", "test3", "test4"]` 
expected is `["test1", "test2", "test3", "test4"]`

**Comment by jdalton** [comment]
Hi @potmo!

Yes, you must provide `sortedUniq` with a sorted array.

**Comment by potmo** [comment]
Shouldn't that be clear in the documentation in that case? It doesn't say anything about it.
It just says that it is optimized for sorted arrays

**Comment by ematipico** [comment]
@potmo The example that you provided has an array that **is not sorted**, so `sortedUniq` won't work. The doc looks fine, doesn't it?

**Comment by potmo** [comment]
I actually did the mistake to first think that `sortedUniq` unique and sorted array.
Then I read the docs and understood it as it was faster for sorted arrays but would work with unsorted arrays. Since I have an array that is usually sorted I went with it and got a bug.
Maybe it is just the docs that is unclear.

[https://lodash.com/docs/4.17.4#sortedUniq](https://lodash.com/docs/4.17.4#sortedUniq)

states that

> This method is like _.uniq except that *it's designed and optimized for sorted arrays.*

but is in reality

> This method is like _.uniq except that *it only works with sorted arrays. But in the case the array is known to be sorted `_.sortedUniq` is faster than `_.uniq`*
